presidio-image-redactor/tests/test_dicom_image_redactor_engine.py

```diff
@@ -1157,6 +1158,126 @@ def test_add_redact_box_happy_path(
+            Path(TEST_DICOM_PARENT_DIR, "0_ORIGINAL.dcm"),
+            True,
+            [PatternRecognizer(supported_entity="PERSON", deny_list=["2"])]
+        ),
+    ],
+)
+def test_get_analyzer_results_happy_path(
+    mocker,
+    mock_engine: DicomImageRedactorEngine,
+    image: Image,
+    dcm_path: str,
+    use_metadata: bool,
+    ad_hoc_recognizers: Optional[List[PatternRecognizer]]
+):
+    """Test happy path for DicomImageRedactorEngine._get_analyzer_results
+
+    Args:
+        mock_engine (DicomImageRedactorEngine): DicomImageRedactorEngine object.
+        image (PIL.Image): A PIL image.
+        dcm_path (pathlib.Path): Path to DICOM file.
+        use_metadata (bool): Whether to consider metadata when running analysis.
+        ad_hoc_recognizers(None or list): Ad-hoc recognizers to use.
+    """
+    # Arrange
+    mock_analyze = mocker.patch(
+        "presidio_image_redactor.dicom_image_redactor_engine.ImageAnalyzerEngine.analyze",
+        return_value=None,
+    )
+    mock_get_text_metadata = mocker.patch(
+        "presidio_image_redactor.dicom_image_redactor_engine.DicomImageRedactorEngine._get_text_metadata",
+        return_value=[None, None, None],
+    )
+    mock_make_phi_list = mocker.patch(
+        "presidio_image_redactor.dicom_image_redactor_engine.DicomImageRedactorEngine._make_phi_list",
+        return_value=None,
+    )
+    mock_pattern_recognizer = mocker.patch(
+        "presidio_image_redactor.dicom_image_redactor_engine.PatternRecognizer",
+        return_value=None,
+    )
+    test_instance = pydicom.dcmread(dcm_path)
+
+    # Act
+    _ = mock_engine._get_analyzer_results(
+        image=image,
+        instance=test_instance,
+        use_metadata=use_metadata,
+        ocr_kwargs=None,
+        ad_hoc_recognizers=ad_hoc_recognizers
+    )
+
+    # Assert
+    if use_metadata is False:
+        mock_analyze.assert_called_once()
+        mock_get_text_metadata.assert_not_called()
+        mock_make_phi_list.assert_not_called()
+        mock_pattern_recognizer.assert_not_called()
+    elif use_metadata is True:
+        mock_analyze.assert_called_once()
+        mock_get_text_metadata.assert_called_once()
+        mock_make_phi_list.assert_called_once()
+        mock_pattern_recognizer.assert_called_once()
+
+@pytest.mark.parametrize(
+    "ad_hoc_recognizers",
+    [
+        ("invalidType"),
+        ([]),
+        ([PatternRecognizer(supported_entity="TITLE", deny_list=["Mr", "Ms"]), 2])
+    ],
+)
+def test_get_analyzer_results_exceptions(
+    mock_engine: DicomImageRedactorEngine,
+    ad_hoc_recognizers: Optional[List[PatternRecognizer]],
+):
+    """Test error handling of DicomImageRedactorEngine _get_analyzer_results()
+
+    Args:
+        mock_engine (DicomImageRedactorEngine): DicomImageRedactorEngine object.
+        ad_hoc_recognizers(None or list): Ad-hoc recognizers to use.
+    """
+    with pytest.raises(Exception):
```

Comment:
Consider checking for `TypeError` and not any exception

Reply:
Will create a separate small PR to make this change such that we can get this current PR merged in.